src/osd/PrimaryLogPG.cc

```diff
@@ -6828,11 +6936,13 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 
 	  if (obs.oi.manifest.is_chunked()) {
 	    src_hoid = obs.oi.soid;
+	    src_hoid.snap = ctx->snapc.seq;
 	    cb = new PromoteManifestCallback(ctx->obc, this, ctx);
 	  } else if (obs.oi.manifest.is_redirect()) {
 	    object_locator_t src_oloc(obs.oi.manifest.redirect_target);
 	    my_oloc = src_oloc;
 	    src_hoid = obs.oi.manifest.redirect_target;
+	    src_hoid.snap = ctx->snapc.seq;
 	    cb = new PromoteManifestCallback(ctx->obc, this, ctx);
```

Comment:
Should this be PromoteCallback?

Reply:
Right. It seems that I made a mistake. This needs finish_promote.